lib/manageiq/cross_repo/runner.rb

```diff
@@ -35,14 +37,32 @@ def run
 
     def run_tests
       with_test_env do
-        system!({"TRAVIS_BUILD_DIR" => test_repo.path.to_s}, "bash", "tools/ci/before_install.sh") if ENV["CI"] && File.exist?("tools/ci/before_install.sh")
-        system!(env_vars, "bin/setup")
-        system!(script_cmd)
+        commands =
+          Array(travis_yml["before_install"]) +
+          Array(travis_yml["install"] || install_cmd) +
+          Array(travis_yml["before_script"]) +
+          Array(travis_yml["script"] || script_cmd) +
+          Array(travis_yml["after_script"])
+
+        template = <<~SCRIPT
+          #!/bin/bash
+
+          set -e
+
+          <% commands.each do |cmd| %>
+          <%= cmd %>
+          <% end %>
+        SCRIPT
+
+        require "erb"
+        bash_script = ERB.new(template).result(binding)
+
+        system!(env_vars, bash_script)
```

Comment:
I didn't realize that system could take a straight script. 😮   You don't have to chmod +x it?

Reply:
It isn't a file so nothing to `chmod`, I'm hoping that the default shell picks up on the shebang and if not I'm going to switch back to executing `bash` but probably tell it to read from stdin with `bash -s`